micrometer-commons/src/main/java/io/micrometer/common/annotation/TagAnnotationHandler.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.common.annotation;
+
+import io.micrometer.common.KeyValue;
+import io.micrometer.common.util.internal.logging.InternalLogger;
+import io.micrometer.common.util.internal.logging.InternalLoggerFactory;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.reflect.MethodSignature;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * This class is able to find all methods annotated with the Micrometer annotations. All
+ * methods mean that if you have both an interface and an implementation annotated with
```

Comment:
Could we add a test to show this works?

Reply:
We have a test for that https://github.com/micrometer-metrics/micrometer/pull/3727/files#diff-f6ba43cbb6d7919b0039dbb5cbef0b4b8b88fa8dbbb959103eb96bb10d2f9a50R46

Reply:
Sorry, my comment wasn't clear at all. I meant to point out the part of the JavaDoc saying this works with both annotated interface and implementations, but the test only uses an annotated concrete class. So it would be nice if we also had a test to show this works with annotations on an interface, and merging the annotations from interface and concrete class implementing the interface.